telemetry-schema.md

```diff
@@ -0,0 +1,71 @@
+## Attributes
+
+Key | Description
+--- | ---
+event | `perf-report` (can this take other values?)
+results | String holding a JSON-encoded `OuterResults`
+sentCount | string-encoded integer representing the number of `Results` in each `Test`
+
+## OuterResults
+
+Key | Description
+--- | ---
+status | `"complete"` (Can this take other values?)
+tests | array of `Test`
+
+## Test
+
+Key | Description
+--- | ---
+ccheck | ?
+doh | `true` or `false`; whether DOH was used to resolve the remote endpoint
+facebook | `true` or missing; whether the remote endpoint is Facebook
+label | friendly slug with a 1:1 mapping to a (`doh`, `url`) pair
+results | array of `Result`
+url | URL of endpoint
+
+## Result
+
+Key | Description
+--- | ---
+asyncOpenTime | timestamp
+channelCreationTime | timestamp
+connectEndTime | timestamp
+connectStartTime | timestamp
+domainLookupEndTime | timestamp
+domainLookupStartTime | timestamp
+errorCode | `0` or missing?
+event | `"loadend"` (Can this take other values?)
+headers | array of `Header`
+isTRR | `true` or missing; whether DOH was used to resolve the remote endpoint
+remoteAddress | IPv[46] address of the fetch endpoint
+requestStartTime | timestamp
+responseCode | HTTP status code for the fetch
+responseEndTime | timestamp
+responseStartTime | timestamp
+secureConnectionStartTime | timestamp or `0`?
+securityState | `2` or missing?
+status | `0`?
+tcpConnectEndTime | timestamp
+
+## Header
+
+Strings matching one of (whitespace added for legibility):
+
+```
+synth-timings-incoming:
+  now=(\d+);
+  turnaround=(\d+);
+  client_rtt=(\d+), now=(\d+);
+  turnaround=(\d+);
```

Comment:
We talked to cloudflare, an apparently turnaround shouldn't be public, and presumably there should only be one client_rtt (there seem to be two sometimes, one of them is probably for the "cache parent")

Reply:
What does the client_rtt measure? What should we do if Cloudflare sends us more than one?

Reply:
It's the round trip time to the client (in this case the browser).
I think we ignored it in the previous experiment anyway, but we should consider using it to put the other timings into perspective.
If they fix it on the cloudflare side, it shouldn't show up twice. If it does, maybe we should ignore it still.